File: app/src/main/java/com/openclassrooms/realestatemanager/database/repositories/VideoDataRepository.java
package com.openclassrooms.realestatemanager.database.repositories;


import androidx.lifecycle.LiveData;

import com.openclassrooms.realestatemanager.database.database.VideoPropertyDao;
import com.openclassrooms.realestatemanager.model.Video_property;

import java.util.List;

public class VideoDataRepository {

    private final VideoPropertyDao videoDao;

    public VideoDataRepository(VideoPropertyDao videoDao) { this.videoDao = videoDao; }

    // --- GET ---

    public LiveData<List<Video_property>> getVideos(){ return this.videoDao.getVideos(); }

    // --- CREATE ---

    public void createVideo(Video_property video){ videoDao.insertVideo(video); }

    // --- DELETE ---
    public void deleteVideo(int videoId){ videoDao.deleteVideo(videoId); }


}

File: app/src/main/java/com/openclassrooms/realestatemanager/database/repositories/VideoDataRepository.kt
package com.openclassrooms.realestatemanager.database.repositories


import androidx.lifecycle.LiveData

import com.openclassrooms.realestatemanager.database.database.VideoPropertyDao
import com.openclassrooms.realestatemanager.model.Video_property

class VideoDataRepository(private val videoDao: VideoPropertyDao) {

    // --- GET ---

    val videos: LiveData<List<Video_property>>
        get() = this.videoDao.videos

    // --- CREATE ---

    fun createVideo(video: Video_property) {
        videoDao.insertVideo(video)
    }

    // --- DELETE ---
    fun deleteVideo(videoId: Int) {
        videoDao.deleteVideo(videoId)
    }


}
